tests/test_association_db_sqlite.py

```diff
@@ -27,21 +27,74 @@
 
 from lsst.ap.association import \
     AssociationDBSqliteTask, \
-    DIAObjectCollection
+    make_minimal_dia_object_schema, \
+    make_minimal_dia_source_schema
 import lsst.afw.image as afwImage
 import lsst.afw.geom as afwGeom
+import lsst.afw.table as afwTable
 import lsst.daf.base as dafBase
 import lsst.pipe.base as pipeBase
 import lsst.utils.tests
-from test_association_task import \
-    create_test_dia_objects, \
-    create_test_dia_sources
+
+
+def create_test_points(n_points=5,
+                       start_id=0,
+                       schema=None,
+                       point_locs_deg=[[0.0, 0.0]],
+                       scatter_arcsec=1.0,
+                       indexer_ids=None,
+                       associated_ids=None):
+    """ Create dummy DIASources or DIAObjects for use in our tests.
+
+    Parameters
+    ----------
+    n_points : `int`
+        Number of fake sources to create for testing.
+    start_id : `int`
+        Unique id of the first object to create. The remaining sources are
+        incremented by one from the first id.
+    schema : `lsst.afw.table.Schema`
+        Schema of the objects to create. Defaults to the DIASource schema.
+    point_locs_deg : `list` of `float`s
+        Positions of the test points to create.
+    scatter_arcsec : `float`
+        Scatter to add to the position of each DIASource.
+    indexer_ids : `list` of `ints`s
+        Id numbers of pixelization indexer to store.
+    associated_ids : `list` of `ints`s
+        Id numbers of associated DIAObjects to store..
```

Comment:
The main thing I use this for is handling the case of real positions here and in AssociationTask where I need to look up realistic pixelization indices using a exposure bounding box and a wcs. This just lets me handle both cases where do or do not need that functionality.

Reply:
I'm confused. Are you talking about a different part of the code?

Reply:
I don't see how having a long parameter list helps with code reuse. It's more things for the caller to juggle/remember (especially since your calls seem to involve magic values), and, in the case of scattering, it violates the Do One Thing principle. How is something like:

    create_test_points(
        n_points=n_objects,
        start_id=0,
        schema=make_minimal_dia_object_schema(),
        point_locs_deg=object_centers,
        scatter_arcsec=1.0,
        indexer_ids=[0] * n_objects,)

more reusable than 

    create_test_points(
        start_id=0,
        schema=make_minimal_dia_object_schema(),
        point_locs_deg=add_scatter(object_centers, 1.0 * arcsec),
        indexer_ids=[0] * n_objects,)

Reply:
Okay, I see what you mean now, I've removed the n_points but kept the scatter_arcsec. Take a look and see if that satisfies.